### ISSUE TITLE
[Bug]: Partially paid and expired invoice with late overpayment

### ISSUE THREAD

**Issue by ndeet** [btcpayserver/btcpayserver#4146]
### What is your BTCPay version?

v1.6.10.0

### How did you deploy BTCPay Server?

btcpayserver-docker

### What happened?

**Following use case:**
- an invoice gets partially paid and expires
- after expiration the invoice is overpaid

### Problem 1: UI
In the invoice details you can't see in any way that the invoice was overpaid as the invoice status is "Expired (PaidLate)" (or I miss something?) 

Shown invoice was over `0.00013790' BTC:
- first payment was `0.00010790` BTC then the invoice expired
- second payment was `0.00010100` BTC

I can't see that this was an overpayment as I do not see the original BTC value anywhere. The amount "Due" should be negative?

Also the invoice status is "Expired" and additional status is "PaidLate" instead of "PaidOver"?

![image](https://user-images.githubusercontent.com/1136761/191248755-34d9fcc1-d764-48eb-b8ca-62b3425a8216.png)

### Problem 2: webhook events

The webhook events of `InvoiceReceivedPayment` and `InvoicePaymentSettled` do also not reflect that it is an overpayment. The field `overPaid` is `false` both times, expected to be `true`.

```json
{
  "afterExpiration": true,
  "paymentMethod": "BTC",
  "payment": {
    "id": "fd8f11483c1804142599cd3ba59e3d456252880f357fe9d2d65792a7851fbbbc-1",
    "receivedDate": 1663672742,
    "value": "0.000101",
    "fee": "0.000001",
    "status": "Invalid",
    "destination": "my7iba8Gxo7WqcwH9H4pD6z176SBKScVsc"
  },
  "overPaid": false,
  "deliveryId": "5xHgTQYuVVaSsFuu8UMyY6",
  "webhookId": "Hy3bxuiiioYwcvjjH7g87",
  "originalDeliveryId": "5xHgTQYuVVaSsFuu8UMyY6",
  "isRedelivery": false,
  "type": "InvoiceReceivedPayment",
  "timestamp": 1663672742,
  "storeId": "4bc71hXipmSvK7e9PAGsRqW61EhtpmATsjcyMu5ko3sL",
  "invoiceId": "YAugfJRpSR7Sea5i1ujgoJ"
}
```

```json
{
  "afterExpiration": true,
  "paymentMethod": "BTC",
  "payment": {
    "id": "5ec9b5fd504dccde8e9f04efe13a12fd000c686c9296502ba462612d1487316b-0",
    "receivedDate": 1663672018,
    "value": "0.0001079",
    "fee": "0.0",
    "status": "Settled",
    "destination": "my7iba8Gxo7WqcwH9H4pD6z176SBKScVsc"
  },
  "overPaid": false,
  "deliveryId": "RnDovMYKE9KEZBvVvZfLYd",
  "webhookId": "Hy3bxuiiioYwcvjjH7g87",
  "originalDeliveryId": "RnDovMYKE9KEZBvVvZfLYd",
  "isRedelivery": false,
  "type": "InvoicePaymentSettled",
  "timestamp": 1663673249,
  "storeId": "4bc71hXipmSvK7e9PAGsRqW61EhtpmATsjcyMu5ko3sL",
  "invoiceId": "YAugfJRpSR7Sea5i1ujgoJ"
}
```

### How did you encounter this bug?

partially pay invoice, let it expire and overpay the remaining amount

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by bolatovumar** [comment]
> Also the invoice status is "Expired" and additional status is "PaidLate" instead of "PaidOver"?

Status can either be `PaidLate` or `PaidOver` and `PaidLate` wins out in this case. I guess we _could_ have a status `PaidLateOver` but that's kind of weird. I wonder if the status shouldn't just be one thing but an array of all things that apply to the invoice.

**Comment by ndeet** [comment]
Not sure about having multiple additional status, maybe @Kukks @dennisreimann has an input to this too. 

Even if it stays as PaidLate at least the "Overpaid" column like with status PaidOver should be visible? 
![image](https://user-images.githubusercontent.com/1136761/191785073-cc584628-5191-4315-a601-55243fb4ba3a.png)

Like it is now you can't see the original BTC amount only "Total Fiat Due". So independent from which status the invoice gets we should at least show:
- the payment method due amount e.g. BTC `0.00013790` in above case
- on summary "Overpaid" column should be visible

**Comment by bolatovumar** [comment]
> Even if it stays as PaidLate at least the "Overpaid" column like with status PaidOver should be visible?

I haven't looked into it but I'm guessing that the same thing is happening here: the "paid late" status "wins out" over the "paid over" status so you don't get this column.

**Comment by ndeet** [comment]
As discussed with @NicolasDorier it is not easily possible to add multiple invoice statuses and the paidLate invoice status is kinda correct even if also overPaid. What we should do is to make the overpayment visible by showing the overpaid column. 

I think generally we should show the initial payment method amount (BTC) after at least one payment arrived with that payment method? At the moment you can't see what the initial BTC amount was, so you have no way to even calculate yourself how much the overpayment was. If this is too complicated because of many payment methods we might want to only show the overpaid column at least.

ping @Kukks for your input

**Comment by ndeet** [comment]
@bolatovumar we discussed in dev call right now. The conclusion was that we should make sure to show the overpaid column on the invoice summary even if the status is `paidLate` (maybe also any other status) so it is always clear there was an overpayment.

**Comment by bolatovumar** [comment]
@ndeet addressed in #4192
